### PR TITLE
Introduced new variants of existing APIs which take serverWaitTime as an additional argument to be used as timeout.

### PR DESCRIPTION
**Pull request by nemakam**
## Description
Introduced new variants of existing APIs which take serverWaitTime as an additional argument to be used as timeout.
ReceiveAsync(),
AcceptMessageSessionAsync()

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/SubscriptionClient.cs [hunk 1]
```diff
@@ -122,22 +122,46 @@ public static SubscriptionClient Create(ServiceBusEntityConnection topicConnecti
...
+        /// </summary>
+        /// <param name="maxMessageCount">The maximum number of messages that will be received.</param>
+        /// <returns>The asynchronous operation.</returns>
         public Task<IList<BrokeredMessage>> ReceiveAsync(int maxMessageCount)
         {
             return this.InnerReceiver.ReceiveAsync(maxMessageCount);
         }
 
+        /// <summary>
+        /// Asynchronously receives a message using the <see cref="MessageReceiver" />.
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/SubscriptionClient.cs]: More of a question - "Asynchronously receives a message". Is "Asynchronously" really needed? The method is suffixed with `Async` and it returns a `Task` (or `Task<T>`). 😃

Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/SubscriptionClient.cs]: PS: applicable to all xmldocs

File: test/Microsoft.Azure.ServiceBus.UnitTests/QueueClientTests.cs [hunk 2]
```diff
@@ -133,5 +133,21 @@ public sealed class QueueClientTests : SenderReceiverClientTestBase
                 await queueClient.CloseAsync();
             }
         }
+
+        [Theory]
+        [MemberData(nameof(TestPermutations))]
+        [DisplayTestMethodName]
+        async Task ServerWaitTimeoutTest(string queueName, int messageCount = 1)
```
Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/QueueClientTests.cs]: `ServerWaitTimeoutTest` doesn't really describe what the outcome of the test is or what's expected. Would rather see a name that implies what the outcome is. E.g. `Should_perform_X`

File: test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs [hunk 3]
```diff
@@ -188,6 +189,28 @@ public sealed class QueueSessionTests
             }
         }
 
+        [Theory]
+        [MemberData(nameof(TestPermutations))]
+        [DisplayTestMethodName]
+        async Task ServerWaitTimeoutTest(string queueName, int messageCount = 1)
```
Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs]: What `ServerWaitTimeoutTest` is doing? Don't force to read the whole test just to decipher what the outcome should be. You don't do that with regular methods, right? 😉

Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs]: @jtaubensee this is probably the argument that I'd use on the unit testing code as well - readability and conveying what tests are doing. Can you look at the name solely and understand what the unit test is trying to achieve? If the answer is "no", then the name is no good.

File: test/Microsoft.Azure.ServiceBus.UnitTests/TopicClientTests.cs [hunk 4]
```diff
@@ -157,5 +157,24 @@ public sealed class TopicClientTests : SenderReceiverClientTestBase
                 await topicClient.CloseAsync();
             }
         }
+
+        [Theory]
+        [MemberData(nameof(TestPermutations))]
+        [DisplayTestMethodName]
+        async Task ServerWaitTimeoutTest(string topicName, int messageCount = 1)
```
Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/TopicClientTests.cs]: no assertions?

File: test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs (hunk 3 above)
Comment (jtaubensee) [inline test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs]: @SeanFeldman - Were you still interested in submitting a PR for the unit test name refactoring?

Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs]: Nicely done hing @jtaubensee 😃 
Yes, still interested. One 🐱 at a time.

Comment (nemakam) [inline test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs]: Updated

File: test/Microsoft.Azure.ServiceBus.UnitTests/QueueClientTests.cs (hunk 2 above)
Comment (nemakam) [inline test/Microsoft.Azure.ServiceBus.UnitTests/QueueClientTests.cs]: Updated

File: src/Microsoft.Azure.ServiceBus/SubscriptionClient.cs (hunk 1 above)
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/SubscriptionClient.cs]: Updated

File: test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs (hunk 3 above)
Comment (SeanFeldman) [inline test/Microsoft.Azure.ServiceBus.UnitTests/QueueSessionTests.cs]: @nemakam sir, thank you from the bottom of my ❤️ for the proper test name. I can die a happy man now. @jtaubensee watch and learn 😛

**Pull request merged**